Source/Concurrency/LinearTypeChecker.cs

```diff
@@ -921,12 +921,12 @@ private class LinearityCheck
       public string message;
       public string name;
 
-      public LinearityCheck(Expr assume, Expr assert, string message, string name)
+      public LinearityCheck(LinearDomain domain, Expr assume, Expr assert, string message, string name)
       {
         this.assume = assume;
         this.assert = assert;
         this.message = message;
-        this.name = name;
+        this.name = $"{domain.domainName}_{name}";
```

Comment:
Maybe store `domain` (or just `domain.domainName`) in `LinearityCheck` and compute the label where the block is created.

Reply:
Done.